Yoast/wpseo-news
# Prepared for PHP 5.2-5.5 drop

## Summary

This PR can be summarized in the following changelog entry:

[More info from same changes on wordpress-seo
](https://github.com/Yoast/wordpress-seo/pull/12129)

In .travis.yml:

* Changed dist from trusty to xenial. But explicitly used 'trusty' in the 5.6 test.
* Added mysql as a service.
* Removed PHP 5.3 and 7.0 from the builds
* Removed multisite from the 7.3 build that doesn't run coverage.
* Using WP_VERSION=latest instead of WP_VERSION=4.9 / WP_VERSION=5.0
* Upgraded PHP 5.2 test from 4.8 to WP_VERSION=4.9
* WordPress Core and wordpress-seo clone depth set from 50 to 1
* Created 2 allowed failures
* Added a script which gets the latest WordPress version

## Relevant technical choices:

* As WPSEO News is not upgraded to PHP 7.3 yet, I could not remove multisite (3rd checkbox in isue).

* I added a scripts which gets the latest WordPress version, `in the before scripts:`:
```
# Careful: The HTTPS version of the following URL is different, therefore we need to use HTTP.
 - |
   if [[ "$WP_VERSION" == "latest" ]]; then
     curl -s http://api.wordpress.org/core/version-check/1.7/ > /tmp/wp-latest.json
     WP_VERSION=$(grep -o '"version":"[^"]*' /tmp/wp-latest.json | sed 's/"version":"//')
   fi 
```

## Test instructions

This PR can be tested by following these steps:

* Travis build doesn't fail
* Travis build doesn't test on PHP5.2.-5.5 anymore.
* No plugin related functionalities should be changed or behave differently.

Fixes #465
